deltachat-ios/Controller/GalleryViewController.swift

```diff
@@ -197,3 +217,112 @@ extension GalleryViewController {
         present(previewController, animated: true, completion: nil)
     }
 }
+
+class GalleryItem {
+
+    var onImageLoaded: ((UIImage?) -> Void)?
+
+    var msg: DcMsg
+
+    var fileUrl: URL? {
+        return msg.fileURL
+    }
+
+    var thumbnailImage: UIImage? {
+        willSet {
+           onImageLoaded?(newValue)
+        }
+    }
+
+    var showPlayButton: Bool {
+        switch msg.viewtype {
+        case .gif, .video:
+            return true
+        default:
+            return false
+        }
+    }
+
+    init(msgId: Int) {
+        self.msg = DcMsg(id: msgId)
+
+        if let key = msg.fileURL?.absoluteString, let image = ThumbnailCache.shared.restoreImage(key: key) {
```

Comment:
ThumnailCache would have to accept `SDAnimatedImage` as values instead of `UIImage`s. `SDAnimatedImage` can be used on any UIImageView and inherits from UIImage

Reply:
Thanks for the input. I changed that (had to replace the `UIImageView` by `SDAnimatedImageView`.